.github/ISSUE_TEMPLATE/07-refactor.yml

```diff
@@ -0,0 +1,63 @@
+name: Refactor (Maintainers)
+description: Used to track refactoring opportunities
+title: "Refactor: "
+labels: ["refactor"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Don't forget to [check for existing refactor issue tickets](https://github.com/ggerganov/llama.cpp/issues?q=is%3Aopen+is%3Aissue+label%3Arefactoring) in case it's already covered.
+        Also you may want to check [Pull request refactor label as well](https://github.com/ggerganov/llama.cpp/pulls?q=is%3Aopen+is%3Apr+label%3Arefactoring) for duplicates too.
+
+  - type: textarea
+    id: background-description
+    attributes:
+      label: Background Description
+      description: Please provide a detailed written description of the pain points you are trying to solve.
+      placeholder: Detailed description behind your motivation to request refactor
+    validations:
+      required: true
+
+  - type: dropdown
+    id: grokkability
+    attributes:
+      label: Whats the current grokkability of the code in question
+      multiple: false
+      options:
+        - Easy
+        - Normal
+        - Hard
+        - "=.0"
+
+  - type: dropdown
+    id: subsystems
+    attributes:
+      label: Which subsystems are you trying to refactor?
+      multiple: true
+      options:
+        - Kompute
+        - Apple Metal
+        - SYCL
+        - Nvidia GPU
+        - Vulkan
+        - Documentation
+        - Testing
+        - Build
+        - Examples
+        - Devops
+        - Python
+        - Script
+        - Android
+        - Server
+        - GGML
+        - Nix
+        - Embedding
```

Comment:
This should probably be free-text instead of dropdown because the list is not exhaustive

Reply:
Hmmm... on second thought best to delete that as it would generally be explained in background anyway